crates/wasmparser/src/validator/operators.rs

```diff
@@ -2715,8 +2747,18 @@ where
         Ok(())
     }
     fn visit_ref_eq(&mut self) -> Self::Output {
-        self.pop_operand(Some(RefType::EQ.nullable().into()))?;
-        self.pop_operand(Some(RefType::EQ.nullable().into()))?;
+        let a = self
+            .pop_maybe_shared_ref(RefType::EQ.nullable())?
+            .expect("an actual ref type");
```

Comment:
Note that `.expect` here isn't valid because that means that this will panic when `ref.eq` is found after unreachable code. The two results here need to be matched and if they're both `Some` then the shared-ness needs to match. If one is `None` or both are `None` then we're in unreachable code and they can be assumed to match (IIRC)